### ISSUE TITLE
Better CSS for printing in the Report Viewer?

### ISSUE THREAD

**Issue by optimocha** [GoogleChrome/lighthouse#13521]
I'm sending HTML files to my clients using the "Save as HTML" button in the Lighthouse Report Viewer menu. I once tried printing the page but both in the summary mode and the expanded mode, the pages look crappy. I used Stylish to style it a bit better and now at least the page breaks work better. This is all I did:

```
@media print {
	.lh-audit-group--load-opportunities,.lh-audit-group--diagnostics,.lh-clump--passed,.lh-footer {page-break-before: always;}
}
```

Obviously it could be even better, but it's a start, right?

**Comment by adamraine** [comment]
I agree print styles could be better. I would personally rather use `page-break-inside: avoid` in some of these places. `page-break-before: always` is more likely to create pages that are mostly blank.

Maybe:
- `page-break-after: always` for categories (performance, accessibility, seo, etc.)
- `page-break-inside: avoid` for audits and some audit groups/clumps

**Comment by optimocha** [comment]
@adamraine makes sense. Also, maybe improve the saved HTML so it can be opened from within Gmail attachments? Right now it gives this error when you click the HTML attachment on Gmail: "Lighthouse report requires JavaScript. Please enable."

**Comment by adamraine** [comment]
> Also, maybe improve the saved HTML so it can be opened from within Gmail attachments? Right now it gives this error when you click the HTML attachment on Gmail: "Lighthouse report requires JavaScript. Please enable."

Gmail strips the JavaScript when opening as an attachment, but the report needs JavaScript to render pretty much everything. I really don't see a way for us to fix this unfortunately. You'll need to download the file from Gmail and then open it.
